scipy/stats/_distribution_infrastructure.py

```diff
@@ -4118,3 +4143,212 @@ def sample(self, shape=(), *, rng=None, method=None):
         x = [var.sample(shape=n, rng=rng) for n, var in zip(ns, self._components)]
         x = np.reshape(rng.permuted(np.concatenate(x)), shape)
         return x[()]
+
+
+class MonotonicTransformedDistribution(TransformedDistribution):
+    r"""Distribution underlying a strictly monotonic function of a random variable
+
+    Given a random variable :math:`X`; a strictly monotonic function
+    :math:`g(u)`, its inverse :math:`h(u) = g^{-1}(u)`, and the derivative magnitude
+    :math: `|h'(u)| = \left| \frac{dh(u)}{du} \right|`, define the distribution
+    underlying the random variable :math:`Y = g(X)`.
+
+    Parameters
+    ----------
+    X : `ContinuousDistribution`
+        The random variable :math:`X`.
+    g, h, dh : callable
+        Elementwise functions representing the mathematical functions
+        :math:`g(u)`, :math:`h(u)`, and :math:`|h'(u)|`
+    logdh : callable, optional
+        Elementwise function representing :math:`\log(h'(u))`.
+        The default is ``lambda u: np.log(dh(u))``, but providing
+        a custom implementation may avoid over/underflow.
+    increasing : bool, optional
+        Whether the function is strictly increasing (True, default)
+        or strictly decreasing (False).
+    g_name : str, optional
+        The name of the mathematical function represented by `g`,
+        used in `__repr__` and `__str__`. The default is ``g.__name__``.
+
+    """
+
+    def __init__(self, dist, *args, g, h, dh, logdh=None,
```

Comment:
It could help with static typing if `dist` could be made positional-only (and also in the superclass). Otherwise I'd have to also account for the situation where there are no `*args` and `dist` is passed by keyword.

Reply:
much appreciated

Reply:
Done in https://github.com/scipy/scipy/pull/21777/commits/47cff617e5530cdf0764a4c7e123fc66753bc378